3.evaluate_model/scripts/nbconverted/get_model_predictions.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### Import Libraries
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import pathlib
+
+from sklearn.metrics import f1_score
+from joblib import load
+
+import sys
+sys.path.append("../utils")
+from split_utils import get_features_data
+from train_utils import get_dataset, get_X_y_data
+
+
+# ### Load necessary data
+
+# In[2]:
+
+
+# load features data from indexes and features dataframe
+data_split_path = pathlib.Path("../1.split_data/indexes/data_split_indexes.tsv")
+data_split_indexes = pd.read_csv(data_split_path, sep="\t", index_col=0)
+features_dataframe_path = pathlib.Path("../0.download_data/data/training_data.csv.gz")
+features_dataframe = get_features_data(features_dataframe_path)
+
+
+# ### Get Each Model Predictions on Each Dataset
+
+# In[3]:
+
+
+# directory to load the models from
+models_dir = pathlib.Path("../2.train_model/models/")
+
+# use a list to keep track of scores in tidy long format for each model and dataset combination
+compiled_predictions = []
+
+# iterate through each model (final model, shuffled baseline model, etc)
+for model_path in models_dir.iterdir():
+    model = load(model_path)
+    model_name = model_path.name.replace("log_reg_","").replace(".joblib","")
+    
+    # iterate through label datasets (labels correspond to train, test, etc)
+    # with nested for loops, we test each model on each dataset(corresponding to a label)
+    for label in data_split_indexes["label"].unique():
```

Comment:
Seeing this inner loop made me wonder if there might be an opportunity to use `pd.DataFrame.groupby()`, `pd.DataFrame.apply()`, or both for function-based output. Using this here may not be necessary but it could prove to be a useful pattern throughout work of this nature.

For ex, from [Pandas groupby docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html#pandas.DataFrame.groupby):

```python
>>> df = pd.DataFrame({'Animal': ['Falcon', 'Falcon',
...                               'Parrot', 'Parrot'],
...                    'Max Speed': [380., 370., 24., 26.]})
>>> df.groupby("Animal", group_keys=True).apply(lambda x: x)
          Animal  Max Speed
Animal
Falcon 0  Falcon      380.0
       1  Falcon      370.0
Parrot 2  Parrot       24.0
       3  Parrot       26.0
```

Reply:
I don't think it would be necessary here but I will consider it for future use, thanks!